src/Microsoft.IdentityModel.Tokens/Experimental/IResultBasedValidation.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace Microsoft.IdentityModel.Tokens.Experimental
+{
+    /// <summary>
+    /// Experimental interface that provides result based token validation instead of throwing exceptions.
+    /// </summary>
+    public interface IResultBasedValidation
+    {
+        /// <summary>
+        /// Validates a token.
+        /// On a validation failure, no exception will be thrown; instead, the <see cref="ValidationError"/> will contain the information about the error that occurred.
+        /// Callers should always check the ValidationResult.IsValid property to verify the validity of the result.
+        /// </summary>
+        /// <param name="token">The token to be validated.</param>
+        /// <param name="validationParameters">The <see cref="ValidationParameters"/> to be used for validating the token.</param>
+        /// <param name="callContext">A <see cref="CallContext"/> that contains call information.</param>
+        /// <returns>A <see cref="ValidationResult{TResult, TError}"/> with either a <see cref="ValidatedToken"/> if the token was validated or an <see cref="ValidationError"/> with the failure information and exception otherwise.</returns>
+        public Task<ValidationResult<ValidatedToken, ValidationError>> ValidateTokenAsync(
+            string token,
+            ValidationParameters validationParameters,
+            CallContext callContext);
```

Comment:
This async API doesn’t accept a cancellation token.
Is it being introduced for backward compatibility with the existing model?

Reply:
@sruke @jmprieur there was some discussion about this where we wanted an API without the cancellation token.
I believe the POR is to add an additional api with a cancellation token, is this understanding correct?

Reply:
Considering there are two public async APIs, one that accepts a cancellation token and one that doesn't (likely as a convenience for those developers who don't need cancellation) and internally we set CancellationToken.None. I don't have a strong preference here and will resolve the conversation.